packages/webview_flutter/webview_flutter_web/lib/src/web_webview_controller.dart

```diff
@@ -118,8 +118,31 @@ class WebWebViewController extends PlatformWebViewController {
   ) async {
     void handler(html.Event event) {
       if (event is html.MessageEvent) {
+        final String? iFrameSrc = _webWebViewParams.iFrame.src;
+        if (event.origin.isEmpty || iFrameSrc == null) {
+          return;
+        }
+
+        // Security check
+        if (!iFrameSrc.startsWith(event.origin)) {
+          return;
+        }
+
+        // ignore: avoid_dynamic_calls
+        final String? channelName = event.data?['channel'] as String?;
+        if (channelName != javaScriptChannelParams.name) {
+          return;
+        }
+
+        // ignore: avoid_dynamic_calls
+        final String? message = event.data?['message'] as String?;
```

Comment:
No need for the explicit variable type, as the compiler can infer it from "`as String?`".

Reply:
This repo has stricter analysis options; I received a number of warnings otherwise.